### PR TITLE
Metabase init always imports new dashboards

### LINKED ISSUE #136: Better import of new canned dashboards
**Is your feature request related to a problem? Please describe.**
When a [new canned dashboard is added to the main line](https://community.faros.ai/docs/canned-dashboards), **existing** users only have a clunky way to have it deployed:
* pull latest code
* archive all canned dashboard
* call the import script that re-adds all dashboards

**Describe the solution you'd like**
At the very least, the ability to import a specific dashboard only.
Better would be the automatic import of new canned dashboards; if an existing canned dashboard has been updated, something (some logic or a human) would need to figure out if an overwrite or duplicate dashboard shall be created.

**Describe alternatives you've considered**
Alternate is the current procedure.

### PR DESCRIPTION
**Pull request by thomas-gerber**
# Description

Metabase init always imports dashboards (`--import-new` option), but only non-existing ones. 

if one uses  the `--import-one` option, the name (with the extension) of a file existing in `resources/metabase/dashboards` is expected. A new dashboard will be created (possibly a named duplicate) in the corresponding collection.

Card behavior is unchanged: all cards needed by created dashboards will be added to the corresponding collection, possibly overwriting cards located them based on the name.

Fixes #136 

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist
- [x] Have you checked to there aren't other open Pull Requests for the same update/change?
- [x] Have you lint your code locally before submission?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Review by ypc-faros — APPROVED** [review]

File: init/scripts/metabase-init.sh [hunk 1]
```diff
@@ -115,9 +115,9 @@ else
...
   fi
-
-  echo "Importing dashboards"
-  node ../lib/metabase/init.js --metabase-url "$mb_url" --username "$mb_user" --password "$mb_password" --database "$db_name" --import
 fi
 
 echo "Attached Faros database to Metabase"
+
+echo "Importing dashboards"
+node ../lib/metabase/init.js --metabase-url "$mb_url" --username "$mb_user" --password "$mb_password" --database "$db_name" --import-new
```
Comment (tovbinm) [inline init/scripts/metabase-init.sh:123]: add a new line

File: init/src/metabase/config.ts [hunk 2]
```diff
@@ -22,3 +22,16 @@ export async function loadDashboards(): Promise<ReadonlyArray<Dashboard>> {
...
   return await Promise.all(promises);
 }
+
+export async function loadDashboard(name: string): Promise<Dashboard> {
+  const dashboardPath = path.join(
+    BASE_RESOURCES_DIR,
+    'metabase',
+    'dashboards',
+    name
+  );
```
Comment (tovbinm) [inline init/src/metabase/config.ts:32]: better:
`const template = await fs.readFile(dashboardPath, 'utf-8')`
and then 
`return { name, template };`

File: init/src/metabase/config.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export interface Dashboard {
   readonly template: string;
 }
 
-export async function loadDashboards(): Promise<ReadonlyArray<Dashboard>> {
+export async function loadDashboards(): Promise<Dashboard[]> {
```
Comment (tovbinm) [inline init/src/metabase/config.ts:11]: what was wrong with `ReadonlyArray<Dashboard>`?

File: init/src/metabase/dashboards.ts [hunk 4]
```diff
@@ -465,8 +465,36 @@ export class Dashboards {
...
 
-  async import(): Promise<void> {
+  async importNew(): Promise<void> {
     const dashboards = await loadDashboards();
+    const existingDashboards = await this.metabase.getDashboards();
+    const existingDashboardNames = new Set<string>();
+
+    for (const item of existingDashboards) {
+      existingDashboardNames.add(item.name);
+      this.logger.info('existing dashboard: %s', item.name);
```
Comment (tovbinm) [inline init/src/metabase/dashboards.ts:475]: `this.logger.debug('Existing dashboard: %s', item.name)`

**Review by tovbinm — APPROVED** [review]
lgtm, some minor comments

**Pull request merged**